droidlet/perception/robot/handlers/detector.py

```diff
@@ -190,12 +192,20 @@ def _maybe_bbox(self, bbox, mask):
 
     def to_struct(self):
         bbox = self._maybe_bbox(self.bbox, self.mask)
+        mask_arr = []
```

Comment:
move this logic to a utility function similar to `_maybe_bbox` - something like `_maybe_mask_points`. it helps to stow away pieces of logic inside functions whenever possible.